File: app/src/main/java/com/volkanhotur/basemvvm/android/ui/screen/main/InformationUseCase.java
package com.volkanhotur.basemvvm.android.ui.screen.main;

import com.volkanhotur.basemvvm.android.data.domain.BaseUseCase;
import com.volkanhotur.basemvvm.android.data.repository.ApiDataSource;

import org.jetbrains.annotations.NotNull;
import org.json.JSONObject;

import java.util.HashMap;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.Observer;
import io.reactivex.functions.Function;
import okhttp3.ResponseBody;

/**
 * @author volkanhotur
 */

public class InformationUseCase extends BaseUseCase<ResponseBody> {

    @Inject
    public InformationUseCase(ApiDataSource apiDataSource) {
        super(apiDataSource);
    }

    @NotNull
    @Override
    public Observable<ResponseBody> buildUseCaseObservable(HashMap<String, Object> params) {
        return getApiDataSource().getRequest("users/JakeWharton");
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/ui/screen/main/InformationUseCase.kt
package com.volkanhotur.basemvvm.android.ui.screen.main

import com.volkanhotur.basemvvm.android.data.domain.BaseUseCase
import com.volkanhotur.basemvvm.android.data.repository.ApiDataSource
import io.reactivex.Observable
import okhttp3.ResponseBody
import java.util.*
import javax.inject.Inject

/**
 * @author volkanhotur
 */
class InformationUseCase @Inject constructor(apiDataSource: ApiDataSource?) : BaseUseCase<ResponseBody?>(apiDataSource!!) {
    override fun buildUseCaseObservable(params: HashMap<String?, Any?>?): Observable<ResponseBody?> {
        return apiDataSource.getRequest("users/JakeWharton")
    }
}